api/src/controllers/projetos.controller.js

```diff
@@ -56,47 +56,63 @@ exports.deleteAll = (req, res) => {
 }
 
 // ações
-exports.groupHorasProjetos = (req, res) => {
+exports.groupHorasProjetos = async (req, res) => {
   const { somaHoras } = projetos
   const { totalPorPeriodo } = horas
   const { inicio, fim } = req.params
-  const idProjeto = parseInt(req.params.id)
+  const ids = JSON.parse(req.params.ids)
 
   let where = {}
 
   if (req.query) {
     where = req.query
   }
+  
+  let arrayHoras = []
 
-  where.projeto = idProjeto
+  try {
+    for (const id of ids) {
+      where.projeto = id
+      await Promise.all([somaHoras(where), totalPorPeriodo(inicio, fim, id)])
+        .then(results => { 
+          const soma = results[0]
+          const periodo = results[1]
 
-  Promise.all([somaHoras(where), totalPorPeriodo(inicio, fim, idProjeto)])
-    .then(results => {
-      const soma = results[0]
-      const periodo = results[1]
+          const horas = parseInt(soma.horas)
+          const extras = parseInt(soma.extras)
+          const totalPeriodo = parseInt(periodo.horas) + parseInt(periodo.extras)
+          
+          if (soma.projeto === null) throw new Error('Projeto sem horas')
 
-      const horas = parseInt(soma.horas)
-      const extras = parseInt(soma.extras)
-      const totalPeriodo = parseInt(periodo.horas) + parseInt(periodo.extras)
-
-      res.send({
-        idProjeto,
-        title: `Horas agrupadas de ${soma.id_projeto.nome}`,
-        horas,
-        extras,
-        total: horas + extras,
-        totalPeriodo
-      })
-    })
-    .catch(err => {
-      res.status(203).send({
-        idProjeto: parseInt(req.params.id),
-        title: `Não há horas cadastradas para este projeto`,
-        message: err.message || 'Ocorreu um erro na busca',
-        horas: 0,
-        extras: 0,
-        total: 0,
-        totalPeriodo: 0
-      })
+          arrayHoras.push({
+            idProjeto: soma.projeto,
+            title: `Horas agrupadas de ${soma.id_projeto.nome}`,
+            horas,
+            extras,
+            total: horas + extras,
+            totalPeriodo
+          })
+        })
+        .catch(err => {
+          if (err.message === 'Projeto sem horas') {
+            arrayHoras.push({
+              idProjeto: id,
+              title: `Não há horas cadastradas para este projeto`,
+              message: err.message || 'Ocorreu um erro na busca',
+              horas: 0,
+              extras: 0,
+              total: 0,
+              totalPeriodo: 0
+            })
+          } else {
+            throw new Error(err)
+          }
+        })
+    }
+    return res.send(arrayHoras)
+  } catch (err) {
+    return res.status(203).send({
```

Comment:
aqui já é um 500

Reply:
a resposta, 203, que estou recebendo:
`Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.`

Reply:
neste caso real deveria ter sido um 403. A requisição no front está errada:

na requisição está
```
/projetos/26/acoes/agrupar-horas/2020-06-01/2020-06-30
```

invés de 
```
projetos/[26]/acoes/agrupar-horas/2020-06-01/2020-06-30
```
> este segundo funciona perfeitamente!